firebase-firestore/src/main/java/com/google/firebase/firestore/local/QueryEngine.java

```diff
@@ -14,17 +14,26 @@
 
 package com.google.firebase.firestore.local;
 
+import androidx.annotation.Nullable;
 import com.google.firebase.database.collection.ImmutableSortedMap;
+import com.google.firebase.database.collection.ImmutableSortedSet;
 import com.google.firebase.firestore.core.Query;
 import com.google.firebase.firestore.model.Document;
 import com.google.firebase.firestore.model.DocumentKey;
 import com.google.firebase.firestore.model.MaybeDocument;
 
-/** Represents a query engine capable of performing queries over the local document cache. */
-interface QueryEngine {
+/**
+ * Represents a query engine capable of performing queries over the local document cache. You must
+ * call setQueryDataProvider() and setLocalDocumentsView() before using.
+ */
+public interface QueryEngine {
+
+  /** Sets the document view to query against. */
+  void setLocalDocumentsView(LocalDocumentsView localDocuments);
```

Comment:
I wish there were a way to avoid this.

Reply:
Yeah, I guess unless we can make the persistence layer just have a notion of the current queue, this is hard to avoid. LGTM unless you want to try your hand at that (probably separately).